sdks/typescript/src/apache_beam/io/pubsub.ts

```diff
@@ -121,9 +121,13 @@ function writeToPubSubRaw(
 
 export function writeToPubSub(topic: string, options: WriteOptions = {}) {
   return async function writeToPubSub(dataPColl: beam.PCollection<Uint8Array>) {
-    return dataPColl //
-      .map((data) =>
-        PubSub.protos.google.pubsub.v1.PubsubMessage.encode({ data }).finish(),
+    return dataPColl
+      .map(
+        withName("writeToPubSubEncode", (data) =>
```

Comment:
@robertwb what is the right name to put here?

Reply:
This is just to make messages, etc. more meaningful to the users. This is good (with the nit that convention is to us CapitalCamelCase for stage names).